src/main/java/org/opensearch/knn/index/codec/nativeindex/QuantizationIndexUtils.java

```diff
@@ -33,7 +33,7 @@ static Object processAndReturnVector(KNNVectorValues<?> knnVectorValues, IndexBu
                 knnVectorValues.getVector(),
                 indexBuildSetup.getQuantizationOutput()
             );
-            return indexBuildSetup.getQuantizationOutput().getQuantizedVector();
+            return indexBuildSetup.getQuantizationOutput().getQuantizedVectorCopy();
```

Comment:
[Not a blocker]
Lets add a java doc on top of this line to tell why we are calling VectorCopy and not `Vector()` function